test/logger.js

```diff
@@ -76,6 +76,58 @@ describe('logger', () => {
     });
   });
 
+  it.only('should censor auth headers', () => {
+    const bundle = {
+      authData: {
+        key: 'verysecret'
+      },
+      headers: {
+        request_headers: {
+          authorization: 'basic dmVyeXNlY3JldA=='
+        },
+        response_headers: {
+          Authorization: 'basic OnZlcnlzZWNyZXRwbGVhc2U='
+        }
+      }
+    };
+    const logger = createlogger({ bundle }, options);
+
+    return logger('123 from url google.com', bundle.headers).then(response => {
+      response.status.should.eql(200);
+      const j = response.content.json;
+      j.data.request_headers.should.eql(
+        'authorization: basic :censored:10:d98440830f:'
+      );
+      j.data.response_headers.should.eql(
+        'Authorization: :censored:30:f914b1b0d1:'
```

Comment:
This is cool! I really like the way that this worked out where the request header is more specific (includes the `basic` part) because it included the auth value while the response header censored the whole thing because it was a sensitive key. 👍 

I was confused as to why the length for the request header was 10 at first (because `dmVyeXNlY3JldA==` is 16 characters). It turns out this must be a side-effect from https://github.com/zapier/zapier-platform-core/pull/151/files#diff-0936f4d83f660e8d3b1ba19e2225e4c6R102 - we're replacing `dmVyeXNlY3JldA==` with the censored version of `verysecret`. I don't think that's inherently bad (we've been doing the same thing for URI-encoded values), just a side-effect to keep in mind. The real key is that the value should be censored, and that's being achieved. 👍

Reply:
awesome look! yeah, you're correct.